examples/boxplot.rs

```diff
@@ -0,0 +1,222 @@
+use itertools::Itertools;
+use plotters::data::fitting_range;
+use plotters::prelude::*;
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+use std::io::{self, prelude::*, BufReader};
+
+fn read_data<BR: BufRead>(reader: BR) -> HashMap<(String, String), Vec<f64>> {
+    let mut ds = HashMap::new();
+    for l in reader.lines() {
+        let line = l.unwrap();
+        let tuple: Vec<&str> = line.split('\t').collect();
+        if tuple.len() == 3 {
+            let key = (String::from(tuple[0]), String::from(tuple[1]));
+            let entry = ds.entry(key).or_insert_with(Vec::new);
+            entry.push(tuple[2].parse::<f64>().unwrap());
+        }
+    }
+    ds
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let root = SVGBackend::new("plotters-doc-data/boxplot.svg", (1024, 768)).into_drawing_area();
+    root.fill(&WHITE)?;
+
+    let (upper, lower) = root.split_vertically(512);
+
+    let args: Vec<String> = env::args().collect();
+
+    let ds = if args.len() < 2 {
+        read_data(io::Cursor::new(get_data()))
+    } else {
+        let file = fs::File::open(&args[1])?;
+        read_data(BufReader::new(file))
+    };
+    let dataset: Vec<(String, String, Quartiles)> = ds
+        .iter()
+        .map(|(k, v)| (k.0.clone(), k.1.clone(), Quartiles::new(&v)))
+        .collect();
+
+    let category = Category::new(
+        "Host",
+        dataset
+            .iter()
+            .unique_by(|x| x.0.clone())
+            .sorted_by(|a, b| b.2.median().partial_cmp(&a.2.median()).unwrap())
+            .map(|x| x.0.clone())
+            .collect(),
+    );
+
+    let mut colors = (0..).map(Palette99::pick);
+    let mut offsets = (-12..).step_by(24);
+    let mut series = HashMap::new();
+    for x in dataset.iter() {
+        let entry = series
+            .entry(x.1.clone())
+            .or_insert_with(|| (Vec::new(), colors.next().unwrap(), offsets.next().unwrap()));
+        entry.0.push((x.0.clone(), &x.2));
+    }
+
+    let values: Vec<f32> = dataset
+        .iter()
+        .map(|x| x.2.values().to_vec())
+        .flatten()
+        .collect();
+    let values_range = fitting_range(values.iter());
+
+    let mut chart = ChartBuilder::on(&upper)
+        .x_label_area_size(40)
+        .y_label_area_size(120)
+        .caption("Ping Boxplot", ("sans-serif", 20).into_font())
+        .build_ranged(
+            values_range.start - 1.0..values_range.end + 1.0,
+            category.range(),
+        )?;
+
+    chart
+        .configure_mesh()
+        .x_desc("Ping, ms")
+        .y_desc(category.name())
+        .y_labels(category.len())
+        .line_style_2(&WHITE)
+        .draw()?;
+
+    for (label, (values, style, offset)) in &series {
+        chart
+            .draw_series(values.iter().map(|x| {
```

Comment:
Seems this gives us some non-determinism in the result (I mean the ordering of series). 
This may make people confusing when run the example and compare the output directly.  
Maybe we can sort the key first?

Reply:
Definitely, thank you for noticing that.